eclipse-scout-core/src/form/fields/listbox/ListBox.ts

```diff
@@ -58,23 +84,45 @@ export class ListBox<TValue> extends LookupBox<TValue> implements ListBoxModel<T
     this.addField(this.table.$container);
   }
 
+  setHideTableLines(hideTableLines: boolean) {
+    this.setProperty('hideTableLines', hideTableLines);
+  }
+
   protected _onTableRowsChecked(event: TableRowsCheckedEvent) {
     this._syncTableToValue();
   }
 
-  protected _syncTableToValue() {
-    if (!this.lookupCall || this._valueSyncing) {
-      return;
-    }
-    this._valueSyncing = true;
-    let valueArray = [];
+  protected _valueToTable(newValue0: TValue) {
+    let opts = {checkOnlyEnabled: false};
+
+    let rowsToCheck = [];
+
+    const newValue = newValue0 as TLookup[];
+    this.table.uncheckRows(this.table.rows, opts);
+    this.table.rows.forEach(row => {
+      if (arrays.containsAny(newValue, row.lookupRow.key)) {
+        rowsToCheck.push(row);
+      }
+    }, this);
+    this.table.checkRows(rowsToCheck, opts);
+  }
+
+  protected _tableToValue(): TValue {
+    const valueArray: TLookup[] = [];
     this.table.rows.forEach(row => {
       if (row.checked) {
         valueArray.push(row.lookupRow.key);
       }
     });
+    return valueArray as TValue;
```

Comment:
Are the constant casts throughout this class really necessary?

Reply:
isolated the method needing casts to a separate section of the class